server/game/cards/08.3-Km/TheSkahazadhan.js

```diff
@@ -0,0 +1,22 @@
+const DrawCard = require('../../drawcard.js');
+
+class TheSkahazadhan extends DrawCard {
+    setupCardAbilities(ability) {
+        this.reaction({
+            when: {
+                onPlotsRevealed: () => true
+            },
+            cost: ability.costs.discardFromHand(),
```

Comment:
Only missing the kneelSelf() cost here (and you could incorporate that in the message as well if you like).